### ISSUE TITLE
Issue with reset in modules

### ISSUE THREAD

**Issue by carlganz** [daattali/shinyjs#133]
Hey Dean,

It appears that there have been issues filed in the past regarding modules so I apologize if this is a duplicate, but I am pretty sure this hasn't been addressed yet.

Essentially I can't reset inputs inside modules, but I can reset values inside of modules from outside of modules (which is not ideal).

MRE:

```{R}
modUI <- function(id) {
  ns <- NS(id)
  tagList(
    textInput(ns("one"), "Testing", "Testing"),
    actionButton(
      ns("two"),
      "Press me to reset value inside module from inside module"
    )
  )
}

mod <- function(input, output, session) {
  observeEvent(input$two, {
    shinyjs::reset(session$ns("one"))
  })
}

library(shiny)
library(shinyjs)

ui <- fluidPage(
  shinyjs::useShinyjs(),
  modUI("test"),
  actionButton(
    "three",
    "Press me to reset value inside module from outside of module"
  )
)

server <- function(input, output, session) {
  callModule(mod, "test")
  
  observeEvent(input$three, {
    shinyjs::reset("test-one")
  })
}

shinyApp(ui = ui, server = server)
```

Change the value of the `textInput`. Hit reset from inside module and nothing happens. Hit rese from outside of module and it works. 

I am guessing that the issue involves environments since `reset` is being called with the same parameter, but getting a different outcome.

**Comment by carlganz** [comment]
Similarly for `toggleState`,

```{R}
modUI <- function(id) {
  ns <- NS(id)
  tagList(
    shinyjs::useShinyjs(),
    textInput(ns("one"), "Testing", "Testing"),
    actionButton(
      ns("two"),
      "Press me to reset value inside module from inside module"
    )
  )
}

mod <- function(input, output, session) {
  observeEvent(input$two, {
    shinyjs::toggleState(session$ns("one"), condition = input$two %% 2)
  })
}

library(shiny)
library(shinyjs)

ui <- fluidPage(
  shinyjs::useShinyjs(),
  modUI("test"),
  actionButton(
    "three",
    "Press me to reset value inside module from outside of module"
  )
)

server <- function(input, output, session) {
  callModule(mod, "test")
  
  observeEvent(input$three, {
    shinyjs::toggleState("test-one", condition = input$three %% 2)
  })
}

shinyApp(ui = ui, server = server)
```

**Comment by daattali** [comment]
So this is actually a true case of "feature, not a bug" :)

It would work if inside the module you **don't** use `ns()`. This is because when modules first came out I had a choice to make: either make shinyjs work with modules only when the ID is 100% correct, or try to be smart about it. I decided to go with the former, because I assumed that most people who use modules would not understand them well enough to realize that the ID in inputs/outputs and the ID in shinyjs are the same. I was getting flooded with emails asking me why it isn't working in modules, people expected it to work without namespacing. When a new user reads the documentation for shiny modules, it's a bit confusing, so they follow the steps exactly which means add ns() to inputs and output. But it seemed like the average user didn't really understand what that means, and that they would need to do the same for shinyjs. So instead of battling that battle, I decided to go the other way and just guess the correct namespace. In hindsight, maybe that wasn't the right solution, I'm not sure. Although to be honest, I've gotten so many emails about modules being broken before I did that, and you're only the second person to contact me about modules being broken BECAUSE of that feature, so I think maybe it was the right choice. 

You are more technically correct than the majority of other users who had the opposite problem, but I'd rather deal with you as a one-off than with all of them :)

I hope my lengthy story made some sense.

BTW this issue is very related to https://github.com/daattali/shinyjs/issues/118
